src/controllers/chess-game-controller.js

```diff
@@ -47,115 +55,189 @@ export default class ChessGameController {
 
-    this.undoModal.requestButton.addEventListener("click", () => {
       const activePlayer = this.game.getActivePlayer();
       const color = activePlayer.color;
-      this.undoModal.hidden = true;
-      handleUndoRequest(color, this.undoModal, this.undoReviewModal)
-    })
-
-    this.undoModal.cancelButton.addEventListener("click", () => {
-      this.undoModal.hidden = true;
-    })
-
-    this.undoReviewModal.acceptButton.addEventListener("click", () => {
-      console.log("accept")
-      this.undoReviewModal.hidden = true;
-      this.undoModal.textarea.value = null
-    })
-
-    this.undoReviewModal.declineButton.addEventListener("click", () => {
-      this.undoReviewModal.hidden = true;
-      this.undoModal.textarea.value = null
-    })
-
-    this.endGameModal.playAgainButton.addEventListener("click", () => {
+      const title = `${color} player wants to undo most recent move. Do you accept?`;
+      displayReviewDialog(
+        this.reviewDialog,
+        title,
+        "Undo move",
+        true,
+        this.undoDialog
+      );
+    });
+
+    this.undoDialog.cancelButton.addEventListener("click", () => {
+      const diag = this.undoDialog.shadowRoot.getElementById("undo-dialog");
+      diag.close();
+    });
+
+    this.endGameDialog.playAgainButton.addEventListener("click", (event) => {
+      event.preventDefault();
+
+      const endDialog =
+        this.endGameDialog.shadowRoot.getElementById("end-dialog");
+      endDialog.close();
       console.log("play again");
     });
 
-    this.endGameModal.exportButton.addEventListener("click", () => {
+    this.endGameDialog.exportButton.addEventListener("click", (event) => {
+      event.preventDefault();
+
       exportToPgn(this.game);
     });
 
-    this.endGameModal.copyButton.addEventListener("click", () => {
+    this.endGameDialog.copyButton.addEventListener("click", (event) => {
+      event.preventDefault();
+
       copyPgn(this.game);
     });
+
+    this.promotionDialog.pieceSelect.addEventListener("change", (event) => {
+      event.preventDefault();
+      this.promotionDialog.acceptButton.value =
+        this.promotionDialog.pieceSelect.value;
+    });
+
+    this.promotionDialog.acceptButton.addEventListener("click", (event) => {
+      event.preventDefault();
+
+      const pawnToPromote = this.game.lastMove.initiatingPiece;
+      const chessPieceElement = this.board.chessPieces.find(
+        (element) => element.piece.id === pawnToPromote.id
+      );
+
+      closePromotionSelect(
+        this.promotionDialog,
+        this.promotionDialog.pieceSelect.value,
+        pawnToPromote
+      );
+      chessPieceElement.remove();
+
+      const promatedPiece = chessPieceElement.piece.promote(
+        this.game.lastMove,
+        this.promotionDialog.pieceSelect.value
+      );
+
+      this.mountSinglePiece(promatedPiece);
+
+      setTimeout(() => {
+        this.rotateBoard()
+        this.game.board.willRotate = true
+      }, 700)
+    });
```

Comment:
the initialize became huge. I can see why you would want to move things out into "activate..." helper methods. A few things:
1. if you want to extract a helper method that involves the same concern, in this case "initialization", a better method name to consider might be `initialize...`
2. I would rather see all of the event handlers extracted and the calls to addEventListener moved into a single function